uk.co.compucorp.civicrm.hrsampledata/CRM/HRSampleData/Upgrader.php

```diff
@@ -183,4 +220,26 @@ private function updateCMSUserEmail($cmsID, $newEmail) {
     // save new user email
     user_save($existingUser, ['mail' => $newEmail]);
   }
+
+  /**
+   * Updates the mapped contact ID in uf_match table
+   * given the CMS user ID and the new contact email
+   *
+   * @param $cmsID
+   * @param $newEmail
+   */
+  private function updateMappedContactID($cmsID, $newEmail) {
+    $newEmailData = civicrm_api3('Email', 'get', [
+      'sequential' => 1,
+      'email' => $newEmail,
+      'limit' => '1',
+    ]);
+    if (!empty($newEmailData['id'])) {
+      $newContactID = $newEmailData['id'];
```

Comment:
This would not be the contact ID but the email ID. We need to get the new `contact_id` from `$newEmailData['values']['contact_id']`;